streams/src/test/java/org/apache/kafka/streams/integration/KafkaStreamsTelemetryIntegrationTest.java

```diff
@@ -0,0 +1,497 @@
+        final Topology topology = topologyType.equals("simple") ? simpleTopology() : complexTopology();
+        /*
+           This test verifies that all Kafka Streams metrics with a thread-id tag get passed to the consumer
+         */
+        try (final KafkaStreams streams = new KafkaStreams(topology, properties)) {
+            streams.start();
+            waitForCondition(() -> KafkaStreams.State.RUNNING == streams.state(),
+                    IntegrationTestUtils.DEFAULT_TIMEOUT,
+                    () -> "Kafka Streams never transitioned to a RUNNING state.");
+
+            final List<MetricName> streamsThreadMetrics = streams.metrics().values().stream().map(Metric::metricName)
+                    .filter(metricName -> metricName.tags().containsKey("thread-id")).collect(Collectors.toList());
+
+            final List<MetricName> streamsClientMetrics = streams.metrics().values().stream().map(Metric::metricName)
+                    .filter(metricName -> metricName.group().equals("stream-metrics")).collect(Collectors.toList());
+
+
+
+            final List<MetricName> consumerPassedStreamThreadMetricNames = INTERCEPTING_CONSUMERS.get(FIRST_INSTANCE_CONSUMER).passedMetrics.stream().map(KafkaMetric::metricName).collect(Collectors.toList());
+            final List<MetricName> adminPassedStreamClientMetricNames = INTERCEPTING_ADMIN_CLIENTS.get(FIRST_INSTANCE_CONSUMER).passedMetrics.stream().map(KafkaMetric::metricName).collect(Collectors.toList());
+
+
+            assertEquals(streamsThreadMetrics.size(), consumerPassedStreamThreadMetricNames.size());
+            consumerPassedStreamThreadMetricNames.forEach(metricName -> assertTrue(streamsThreadMetrics.contains(metricName), "Streams metrics doesn't contain " + metricName));
+
+            assertEquals(streamsClientMetrics.size(), adminPassedStreamClientMetricNames.size());
+            adminPassedStreamClientMetricNames.forEach(metricName -> assertTrue(streamsClientMetrics.contains(metricName), "Client metrics doesn't contain " + metricName));
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("multiTaskParameters")
+    @DisplayName("Correct streams metrics should get passed with dynamic membership")
+    public void shouldPassCorrectMetricsDynamicInstances(final boolean stateUpdaterEnabled) throws InterruptedException {
+        final Properties properties1 = props(stateUpdaterEnabled);
+        properties1.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory(appId).getPath() + "-ks1");
+        properties1.put(StreamsConfig.CLIENT_ID_CONFIG, appId + "-ks1");
+
+
+        final Properties properties2 = props(stateUpdaterEnabled);
+        properties2.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory(appId).getPath() + "-ks2");
+        properties2.put(StreamsConfig.CLIENT_ID_CONFIG, appId + "-ks2");
+
+        /*
+          This test ensures metrics are registered and removed correctly with Kafka Steams dynamic membership changes
```

Comment:
What is the value of this comment? Seem `@DisplayName` describe it already?

Reply:
removed